src/dmd/expressionsem.d

```diff
@@ -3013,13 +3016,19 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 // overload of opCall, therefore it's a call
                 if (exp.e1.op != TOK.type)
                 {
-                    if (sd.aliasthis && exp.e1.type != exp.att1)
+                    if (!exp.aliasthislock) //we don't need the recursion. On recursion done in iterateAliasThis
                     {
-                        if (!exp.att1 && exp.e1.type.checkAliasThisRec())
-                            exp.att1 = exp.e1.type;
-                        exp.e1 = resolveAliasThis(sc, exp.e1);
-                        goto Lagain;
+                        exp.aliasthislock = true;
+                        Expression[] results;
+                        iterateAliasThis(sc, exp.e1, &UnaAliasThisCtx(exp).atUna, results);
+                        Expression ret = enforceOneResult(results, exp.loc, "unable to unambiguously resolve %s; candidates:", exp.toChars());
```

Comment:
I see this error message displayed multiple times in this PR, with accompanying `iterateAliasThis`.  Is it possible to reduce the code duplication in a nested function or something similar?  With it located in one logical block, it would also open the opportunity to document it.  It may not be worth it though; it's just a suggestion.